### ISSUE TITLE
Parsing of arguments with optional value is broken

### ISSUE THREAD

**Issue by imbrish** [thephpleague/climate#122]
Following snippet:

```php
$climate = new CLImate;

$climate->arguments->add([
    'foo' => [
        'prefix' => 'f',
    ],
    'bar' => [
        'prefix' => 'b',
    ],
]);

$climate->arguments->parse();

var_dump($climate->arguments->toArray());
```

When called with `command -f -b abc` will yield:

```
array(2) {
  ["foo"]=>
  string(2) "-b"
  ["bar"]=>
  string(3) "abc"
}
```

Looks like the value of `foo` is wrong, despite that `bar` was parsed correctly.

**Comment by duncan3dc** [comment]
Fixed by @mark-gerarts in #133

**Comment by duncan3dc** [comment]
This has been fixed in version [3.5.0](https://github.com/thephpleague/climate/releases/tag/3.5.0)
